Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ComponentMode/ComponentModeDelegate.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/EditorComponentBase.h>
+#include <AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+
+#pragma optimize("", off)
+#pragma inline_depth(0)
+namespace AzToolsFramework::ComponentModeFramework
+
+{
+    // ComponentData constructor to fill in component info
+    ComponentData::ComponentData(AZ::EntityComponentIdPair pairId)
+        : m_pairId(pairId)
+    {
+        AZ::ComponentApplicationBus::BroadcastResult(
+            m_entity, &AZ::ComponentApplicationBus::Events::FindEntity, AZ::EntityId(pairId.GetEntityId()));
+        AZ::ComponentId componentId = pairId.GetComponentId();
+
+        m_component = m_entity->FindComponent(componentId);
+
+        m_componentName = AzToolsFramework::GetFriendlyComponentName(m_component);
+
+        const auto componentTypeId = m_component->GetUnderlyingComponentType();
+
+        AZStd::string iconStr;
```

Comment:
update: Doesn't look like we need this local variable as we're writing to the member variable